mithril-common/src/chain_observer/cli_observer.rs

```diff
@@ -448,8 +448,7 @@ impl ChainObserver for CardanoCliChainObserver {
     ) -> Result<Option<KESPeriod>, ChainObserverError> {
         let dir = std::env::temp_dir().join("mithril_kes_period");
```

Comment:
We could probably have used the new `TmpDir` tool here?

Reply:
This is production code, the `TmpDir` api was only designed to unify test tmp folders generation. We can generalize it but this will needs some more care for the folders names.